### PR TITLE
Propagate build time constants for the kubectl package

### LINKED ISSUE #880: k0s kubectl version not showing correct client version
**Version**
```
$ k0s version
v1.21.0+k0s.0
```

Also observed on `v1.20.5-k0s1`.

**Platform**
Which platform did you run k0s on?
```
$ lsb_release -a
# not available on systems
```
**What happened?**

Output of `k0s kubectl version` does not show correct client version. See 'logs' section for detailed output.

Extract from output to illustrate issue:

```
version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"$Format:%H$",
```

**How To Reproduce**

Run `k0s kubectl version`.

**Expected behavior**

Correct client version in output.

**Screenshots & Logs**

```
$ ./k0s-v1.21.0+k0s.0-amd64 --data-dir $PWD/k0s-data kubectl version
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"$Format:%H$",  \
  GitTreeState:"", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.0-k0s1", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", \
  GitTreeState:"clean", BuildDate:"2021-04-29T15:07:22Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by soider**
**Issue**
Fixes #880

**What this PR Includes**
Propagate ld flags for the kubectl version subcommand.

### CODE REVIEW

**Comment by jnummelin** [comment]
hmm, I think the `k0s kc version` output is bit wrong:
```
$ ./k0s kc version --client=true
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.20.2", GitCommit:"not_available", GitTreeState:"", BuildDate:"2021-05-07T07:37:53Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
```

This comes from the kube go mod versioning detial where the modules are versioned as 0.20.5 even when it actually builds into 1.20.5 kube bins.

I _think_ the output should be something like:
```
$ ./k0s kc version --client=true
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"not_available", GitTreeState:"", BuildDate:"2021-05-07T07:37:53Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by soider** [comment]
Fixed.
Now the output is ```sudo ./k0s kc version --client
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"not_available", GitTreeState:"", BuildDate:"2021-05-11T10:03:26Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}```

According to the KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/917-go-modules it it safe to just replace "v0" to "v1" in the version tag.

File: Makefile [hunk 1]
```diff
@@ -30,6 +35,12 @@ LD_FLAGS += -X github.com/k0sproject/k0s/pkg/build.EtcdVersion=$(etcd_version)
 LD_FLAGS += -X github.com/k0sproject/k0s/pkg/build.KonnectivityVersion=$(konnectivity_version)
 LD_FLAGS += -X \"github.com/k0sproject/k0s/pkg/build.EulaNotice=$(EULA_NOTICE)\"
 LD_FLAGS += -X github.com/k0sproject/k0s/pkg/telemetry.segmentToken=$(SEGMENT_TOKEN)
+LD_FLAGS += -X k8s.io/component-base/version.gitVersion="v$(KUBECTL_VERSION)"
```
Comment (kke) [inline Makefile:38]: If these are only used for the kubectl version, maybe the variable names should be something like `kubectlGitVersion`?

Comment (soider) [reply Makefile:38]: The constants are coming from 3rd party package, unfortunately, so we can't control naming and the env variables already have KUBECTL_ prefix (apart from the BUILDDATE which in my opinion generic one)

**Review by jasmingacic — APPROVED** [review]
Works as expected 
```
./k0s kc version --client=true
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"not_available", GitTreeState:"", BuildDate:"2021-05-17T08:20:02Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Pull request merged**